lib/Agrammon/OutputFormatter/JSON.pm6

```diff
@@ -93,10 +93,25 @@ sub make-record($fq-name, $output, $model, $raw-value, $var, :$language, :$print
 sub push-filters(@records, $fq-name, $output, $model,
                  Agrammon::Outputs::FilterGroupCollection $collection,
                  $var) {
+    # model filter groups
+    my %fgs = $model.filter-groups;
+
+    # add existing filter groups
     for $collection.results-by-filter-group {
         my %filters := .key;
         my $value   := .value;
         push @records, make-record($fq-name, $output, $model, $value, $var, :%filters);
+        for %filters.kv -> $key, $value {
+            # remove from model filter group
+            %fgs{$key}{$value}:delete;
+        }
+    }
+
+    # add remaining model filter groups
+    for %fgs.kv -> $fg, %values {
+        for %values.kv -> $key, $value {
+            push @records, make-record($fq-name, $output, $model, Nil, $var, :filters( %( $fg => $key) ) );
+        }
```

Comment:
So this could end up with entries for filter values that have nothing to do with the module that the filters originated in?

Reply:
The filter groups are being used to "attribute" outputs from "anywhere" in the model to the filter groups. E.g., the fluxes and emissions from storage are attributed to the elements of the `animalcategory` filter group.
There is a requirement to show the distribution of those outputs to **all** animal categories (even those not occurring in a given dataset and thus being 0) always in the same order to ease post-processing.

I am not sure how this extends to multiple filter groups. But I guess it would work similarly for example for multiple storage types (if those existed) and one could attribute outputs to those as well or even to both filter groups (by storage type and animal category) and those could be listed similarly in a report showing "empty slots" as well.